client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts

```diff
@@ -20,26 +20,39 @@ import {
 import { requestPlans } from 'state/plans/actions';
 import { getPlanBySlug, getPlans, isRequestingPlans } from 'state/plans/selectors';
 import { getProductsList, isProductsListFetching } from 'state/products-list/selectors';
-import { requestProductsList } from 'state/products-list/actions';
 import getUpgradePlanSlugFromPath from 'state/selectors/get-upgrade-plan-slug-from-path';
-import { createItemToAddToCart } from './add-items';
+import { createItemToAddToCart } from '../add-items';
+import { RequestCartProduct } from '../types/backend/shopping-cart-endpoint';
 
 const debug = debugFactory( 'calypso:composite-checkout:use-prepare-product-for-cart' );
 
+interface PreparedProductsForCart {
```

Comment:
Should this be exported or moved into a types file?

Reply:
I don't know. What's the benefit of doing so?

Reply:
Filtering out "empty" items is a pretty common practice. You could extend this even further to exclude any "falsy" value in the type guard.

Reply:
🤔  Was that reply meant for the following comment?